syntax = "proto3";

message Empty{

}

message Numbers{
    float x1 = 1;
    float x2 = 2;
    float x3 = 3;
    float x4 = 4;
    float x5 = 5;
    float x6 = 6;
    float x7 = 7;
    float x8 = 8;
}

message Number {
    float value = 1;
}

message String {
    string word = 1;
}

service Calculator {
    rpc SquareRoot(Number) returns (Number) {}
    rpc UpperCase(String) returns (String) {}
    rpc SquareEight(Numbers) returns (Number) {}
    rpc ExecVoid(Empty) returns (Empty) {}
}